internal/api/external.go

```diff
@@ -218,11 +214,30 @@ func (a *API) internalExternalProviderCallback(w http.ResponseWriter, r *http.Re
 		}
 		return nil
 	})
+
+	rurl := a.getExternalRedirectURL(r)
 	if err != nil {
+		msg := ""
+		if errors.Is(err, errEmailVerificationRequired) {
+			msg = fmt.Sprintf("Unverified email with %v. Verify the email with %v in order to sign in", providerType, providerType)
+		} else if errors.Is(err, errEmailConfirmationSent) {
+			msg = fmt.Sprintf("Unverified email with %v. A confirmation email has been sent to your %v email", providerType, providerType)
+		}
+
+		if errors.Is(err, errEmailVerificationRequired) || errors.Is(err, errEmailConfirmationSent) {
```

Comment:
decided to update the `db.Transaction` function to deal with these kind of errors where we want to transaction to commit instead of rolling back in another PR